cfn-resources/cluster/cmd/resource/resource.go

```diff
@@ -23,446 +41,261 @@ func castNO64(i *int64) *int {
 		return handler.ProgressEvent{
 			OperationStatus:  handler.Failed,
 			Message:          err.Error(),
 			HandlerErrorCode: cloudformation.HandlerErrorCodeServiceInternalError}, nil
 	}
 
-	_, _ = logger.Debugf("Created new deployment secret for cluster. Secert Name = Cluster Id:%s", *secretName)
+	_, _ = log.Warnf("Created new deployment secret for cluster. Secert Name = Cluster Id:%s", *secretName)
 	currentModel.Id = secretName
-
-	one := int64(1)
-	three := int64(3)
-	replicationFactor := &three
-	if currentModel.ReplicationFactor != nil {
-		rf := int64(*currentModel.ReplicationFactor)
-		replicationFactor = &rf
-	} else {
-		_, _ = logger.Debugf("Default setting ReplicationFactor to 3")
+	var none = "NONE"
+	if currentModel.EncryptionAtRestProvider == nil {
+		currentModel.EncryptionAtRestProvider = &none
+	}
+	// Atlas client
+	clusterRequest := &mongodbatlas.AdvancedCluster{
+		Name:                     *currentModel.Name,
+		EncryptionAtRestProvider: *currentModel.EncryptionAtRestProvider,
+		ReplicationSpecs:         expandReplicationSpecs(currentModel.ReplicationSpecs),
 	}
 
-	numShards := &one
-	if currentModel.NumShards != nil {
-		ns := int64(*currentModel.NumShards)
-		numShards = &ns
-	} else {
-		_, _ = logger.Debugf("Default setting NumShards to 1")
+	if currentModel.EncryptionAtRestProvider != nil {
+		clusterRequest.EncryptionAtRestProvider = *currentModel.EncryptionAtRestProvider
 	}
 
-	clusterRequest := &mongodbatlas.Cluster{
-		Name:                     cast.ToString(currentModel.Name),
-		EncryptionAtRestProvider: cast.ToString(currentModel.EncryptionAtRestProvider),
-		ClusterType:              cast.ToString(currentModel.ClusterType),
-		ReplicationFactor:        replicationFactor,
-		NumShards:                numShards,
+	if currentModel.ClusterType != nil {
+		clusterRequest.ClusterType = *currentModel.ClusterType
 	}
 
 	if currentModel.BackupEnabled != nil {
 		clusterRequest.BackupEnabled = currentModel.BackupEnabled
 	}
 
-	if currentModel.ProviderBackupEnabled != nil {
-		clusterRequest.ProviderBackupEnabled = currentModel.ProviderBackupEnabled
+	if currentModel.BiConnector != nil {
+		clusterRequest.BiConnector = expandBiConnector(currentModel.BiConnector)
 	}
 
 	if currentModel.DiskSizeGB != nil {
-		currentModel.DiskSizeGB = clusterRequest.DiskSizeGB
+		clusterRequest.DiskSizeGB = currentModel.DiskSizeGB
+	}
+
+	if len(currentModel.Labels) > 0 {
+		clusterRequest.Labels = expandLabelSlice(currentModel.Labels)
+		if containsLabelOrKey(clusterRequest.Labels, defaultLabel) {
+			_, _ = log.Warnf("Create - error: %+v", err)
+			return progress_events.GetFailedEventByCode(
+				LabelError,
+				cloudformation.HandlerErrorCodeInvalidRequest), nil
+		}
 	}
 
 	if currentModel.MongoDBMajorVersion != nil {
 		clusterRequest.MongoDBMajorVersion = formatMongoDBMajorVersion(*currentModel.MongoDBMajorVersion)
 	}
 
-	if currentModel.BiConnector != nil {
```

Comment:
Can we isolate this currentmodel property assigment in a new function, i would make the Create function more clean

Reply:
Fixed